packages/expo-notifications/android/src/main/java/expo/modules/notifications/channels/AndroidAwareChannelManager.java

```diff
@@ -0,0 +1,93 @@
+package expo.modules.notifications.channels;
+
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.content.Context;
+import android.os.Build;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+
+public class AndroidAwareChannelManager  implements ChannelManager {
+
+  ChannelManager nextChannelManager = new PersistentChannelManager();
+
+  @Override
+  public void addChannel(String channelId, ChannelPOJO channel, final Context context) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      NotificationChannel newChannel = new NotificationChannel(channelId, channel.getChannelName(), channel.getImportance());
+
+      if (channel.getChannelId() != null) {
+        newChannel.setGroup(channel.getChannelId());
+      }
+
+      if (channel.getDescription() != null) {
+        newChannel.setDescription(channel.getDescription());
+      }
+
+      newChannel.setVibrationPattern(channel.getVibrate());
+
+      if (!channel.getSound()) {
+        newChannel.setSound(null, null);
+      }
+
+      newChannel.setShowBadge(channel.getBadge());
+      NotificationManager notificationManager = context.getSystemService(NotificationManager.class);
+      notificationManager.createNotificationChannel(newChannel);
+    } else {
+      nextChannelManager.addChannel(channelId, channel, context);
```

Comment:
Instead of making this strange pattern looking like partly chaining and partly inheritance, couldn't we instantiate only the ChannelManager that is applicable to current platform?